llvm/unittests/Support/KnownBitsTest.cpp

```diff
@@ -559,6 +559,42 @@ TEST(KnownBitsTest, BinaryExhaustive) {
       KnownBits::mulhu,
       [](const APInt &N1, const APInt &N2) { return APIntOps::mulhu(N1, N2); },
       checkCorrectnessOnlyBinary);
+
+  testBinaryOpExhaustive(
+      [](const KnownBits &Known1, const KnownBits &Known2) {
+        return KnownBits::avgFloorS(Known1, Known2);
+      },
+      [](const APInt &N1, const APInt &N2) {
+        return APIntOps::avgFloorS(N1, N2);
+      },
+      checkCorrectnessOnlyBinary);
```

Comment:
You shouldn't need lambdas for these:
```cpp
testBinaryOpExhaustive(KnownBits::avgFloorS, APIntOps::avgFloorS, checkCorrectnessOnlyBinary);
```

Reply:
What values are there not complete for?

Reply:
I think you could probably make the KnownBits implementation optimal by using the ext/add/shift/trunc method.